Update docs for SHS and Spark UI

alembiewski opened the pull request:
## Jira Ticket
[DCOS-57164 [MJS] Add documentation for enabling Driver/History UI](https://jira.mesosphere.com/browse/DCOS-57164)

## Description of changes being made

- Updated SHS configuration section
- Added `Spark UI` section with a basic description

akirillov commented:
soft ping @SarahStegall for the review of this part

samvantran commented on pages/services/data-science-engine/1.0.0/hdfs/index.md:
```diff
@@ -75,7 +75,7 @@ dcos {{ model.packageName }} run --submit-args="\
 # Enabling the Spark History Server
 
-{{ model.techName }} includes the Spark History Server. Because the history server requires HDFS, you must explicitly enable it.
+{{ model.techName }} includes the Spark History Server, which is enable by default. It is highly recommended to use HDFS as a storage for SHS and Spark event log. 
```
```suggestion
{{ model.techName }} includes the Spark History Server (SHS), which is enabled by default. It is highly recommended to use HDFS as the backend storage for SHS.
```

samvantran commented on pages/services/data-science-engine/1.0.0/spark/index.md:
```diff
@@ -195,3 +195,6 @@ dcos {{ model.packageName }} run --submit-args="--conf {{ model.packageName }}.m
 The default {{ model.techName }} distribution is compiled against Hadoop 2.9 libraries.  
+
+# Spark UI
+Spark Web UI starts automatically when SparkContext is created and is available at `http://<dcos_url>/service/{{ model.serviceName }}/sparkui`.
```
```suggestion
Spark Web UI starts automatically when a SparkContext is created and is available at `http://<dcos_url>/service/{{ model.serviceName }}/sparkui`.
```

samvantran commented on pages/services/data-science-engine/1.0.0/hdfs/index.md:
```diff
@@ -94,48 +94,28 @@ docker run -it mesosphere/hdfs-client:1.0.0-2.6.0 bash
-"config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints"
-}
+  "spark": {
+   "spark_eventlog_enabled": true,
```
```suggestion
    "spark_eventlog_enabled": true,
```

samvantran replied:
Your JSON is slightly off with only one space between levels. I added one extra one.

Btw, can we group all the configs together since they all fall under `spark`? (including the one [above](https://github.com/mesosphere/dcos-docs-site/pull/2456/files#diff-1a953ce258ff1442629ba630b861c0cdR102))

alembiewski replied:
Yes, sounds good! I made them separate just to show exact properties that need to be configured for each step. I updated the second code snippet to include complete required configuration.

samvantran commented on pages/services/data-science-engine/1.0.0/hdfs/index.md:
```diff
@@ -94,48 +94,28 @@ docker run -it mesosphere/hdfs-client:1.0.0-2.6.0 bash
-}
+  "spark": {
+   "spark_eventlog_enabled": true,
+   "spark_eventlog_dir": "hdfs://hdfs/history"
```
```suggestion
    "spark_eventlog_dir": "hdfs://hdfs/history"
```

samvantran commented on pages/services/data-science-engine/1.0.0/hdfs/index.md:
```diff
@@ -94,48 +94,28 @@ docker run -it mesosphere/hdfs-client:1.0.0-2.6.0 bash
-   "hdfs-config-url": "http://api.hdfs.marathon.l4lb.thisdcos.directory/v1/endpoints"
-              }
+  "spark": {
+   "spark_history_fs_logdirectory": "hdfs://hdfs/history"
```
```suggestion
    "spark_history_fs_logdirectory": "hdfs://hdfs/history"
```

alembiewski commented:
@samvantran, @akirillov thank you for the review! I updated the PR with few commits, please take a look. 

@akirillov, I added documentation about default config (with FS) and HDFS. Regarding S3 configuration example for SHS - after quick googling, it looks like it is will not work without additional configuration and, probably need to be implemented in scope of a separate task.
